pkg/credentialprovider/aws/aws_credentials.go

```diff
@@ -17,25 +17,68 @@ limitations under the License.
 package credentials
 
 import (
+	"bytes"
 	"encoding/base64"
+	"errors"
 	"fmt"
+	"io/ioutil"
+	"net/http"
 	"strings"
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
 	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ecr"
 	"k8s.io/klog"
 
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/kubernetes/pkg/credentialprovider"
 	"k8s.io/kubernetes/pkg/version"
```

Comment:
i know this is unrelated, but can we get rid of the use of `k8s.io/kubernetes/pkg/version`? you can use [`DefaultKubernetesUserAgent`](https://cs.k8s.io/?q=DefaultKubernetesUserAgent&i=nope&files=&repos=kubernetes/kubernetes)  Another PR for that change is ok with me as well.

Reply:
separate PR is better for that